src/DynamoCoreWpf/UI/GuidedTour/GuidesValidationMethods.cs

```diff
@@ -7,13 +13,119 @@ namespace Dynamo.Wpf.UI.GuidedTour
                 termsOfServiceAccepted = dynViewModel.Model.PreferenceSettings.PackageDownloadTouAccepted;
-            return termsOfServiceAccepted; 
+            return termsOfServiceAccepted;
+        }
+
+        /// <summary>
+        /// This method will be executed when passing from Step 2 to Step 3 in the Packages guide, so it will show the TermsOfUse Window in case it was not accepted yet
+        /// </summary>
+        /// <param name="stepInfo"></param>
+        /// <param name="uiAutomationData"></param>
+        /// <param name="enableFunction"></param>
+        /// <param name="currentFlow"></param>
+        internal static void ExecuteTermsOfServiceFlow(Step stepInfo, StepUIAutomation uiAutomationData, bool enableFunction, GuideFlow currentFlow)
+        {
+            CurrentExecutingStep = stepInfo;
+
+            //When enableFunction = true, means we want to show the TermsOfUse Window (this is executed in the UIAutomation step in the Show() method)
+            if (enableFunction)
+            {
+                //If the TermsOfService is not accepted yet it will show the TermsOfUseView otherwise it will show the PackageManagerSearchView
+                stepInfo.DynamoViewModelStep.ShowPackageManagerSearch(null);
+                Window ownedWindow = Guide.FindWindowOwned(stepInfo.HostPopupInfo.WindowName, stepInfo.MainWindow as Window);
+                if (ownedWindow == null) return;
+                Button buttonElement = Guide.FindChild(ownedWindow, stepInfo.HostPopupInfo.HostUIElementString) as Button;
+
+                //When the Accept button is pressed in the TermsOfUseView then we need to move to the next Step
+                if (buttonElement != null)
+                    buttonElement.Click += ButtonElement_Click;
+            }
+            //When enableFunction = false, means we are hiding (closing) the TermsOfUse Window due that we are moving to the next Step or we are exiting the Guide
+            else
+            {
+                Window ownedWindow = Guide.FindWindowOwned(stepInfo.HostPopupInfo.WindowName, stepInfo.MainWindow as Window);
+                if (ownedWindow == null) return;
+                Button buttonElement = Guide.FindChild(ownedWindow, stepInfo.HostPopupInfo.HostUIElementString) as Button;
+                if (buttonElement != null)
+                    buttonElement.Click -= ButtonElement_Click;
+
+                //Tries to close the TermsOfUseView or the PackageManagerSearchView if they were opened previously
+                Guide.CloseWindowOwned(stepInfo.HostPopupInfo.WindowName, stepInfo.MainWindow as Window);
+            }
+        }
+
+        /// <summary>
+        /// This method will be executed when the Accept button is pressed in the TermsOfUseView Window
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private static void ButtonElement_Click(object sender, RoutedEventArgs e)
```

Comment:
@QilongTang I've talked with @filipeotero about it since is working on the ExitGuidePopup that appears when exiting the Guide or when pressing the Decline button in the TermsOfService Window, then I will be modifying this method to be specific for the Accept button. In the future when we implement the functionality for the Decline button probably this  method will be updated.
Then is fixed in the next commit: https://github.com/DynamoDS/Dynamo/pull/12202/commits/22a7e821b48fe97ef45fc79327748d84c9c76df9
Thanks

Reply:
Ok, thanks @RobertGlobant20